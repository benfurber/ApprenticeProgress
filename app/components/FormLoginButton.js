import React from "react";
import { AsyncStorage, StyleSheet } from "react-native";
import { graphql } from "react-apollo";
import { Button, Text, View } from "native-base";

import { loginMutation } from "queries";
import { colours } from "styles";
import { handlePress } from "utils";

const FormLoginButtonUnwrapped = props => {
  const { email, mutate, password } = props;

  return (
    <View style={styles.formField}>
      <Button
        onPress={() => {
          mutate({
            variables: {
              email,
              password,
            },
          });
        }}
        style={styles.button}
      >
        <Text>Login</Text>
      </Button>
    </View>
  );
};

const styles = StyleSheet.create({
  button: {
    flex: 1,
    backgroundColor: colours.red,
  },
  formField: {
    paddingLeft: 15,
    paddingVertical: 10,
  },
});

const options = props => ({
  onCompleted: async data => {
    const token = data.loginUser.user.authenticationToken;
    await AsyncStorage.setItem("token", token);
    return handlePress("Main", props.navigation)();
  },
});

const FormLoginButton = graphql(loginMutation, { options })(
  FormLoginButtonUnwrapped
);

export { FormLoginButton, FormLoginButtonUnwrapped };
