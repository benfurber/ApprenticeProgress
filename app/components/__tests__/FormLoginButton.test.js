// @flow

import React from "react";
import { shallow } from "enzyme";

import { Button } from "native-base";

import { FormLoginButtonUnwrapped } from "../FormLoginButton";

describe("Loading", () => {
  it("has a valid snapshot", () => {
    const mutateMock = () => jest.fn();
    const component = shallow(
      <FormLoginButtonUnwrapped
        mutate={mutateMock}
        email="test@test.com"
        password="WELLSECURE1"
      />
    );

    expect(component).toMatchSnapshot();
  });

  it("mutate calls the other props on press", () => {
    const mutateMock = jest.fn();
    const email = "test@test.com";
    const password = "WELLSECURE1";

    const component = shallow(
      <FormLoginButtonUnwrapped
        mutate={mutateMock}
        email={email}
        password={password}
      />
    );

    component.find(Button).simulate("press");

    expect(mutateMock).toHaveBeenCalledWith({ variables: { email, password } });
  });
});
