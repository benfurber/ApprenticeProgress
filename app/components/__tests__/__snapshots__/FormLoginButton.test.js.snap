// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading has a valid snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FormLoginButtonUnwrapped
    email="test@test.com"
    mutate={[Function]}
    password="WELLSECURE1"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Styled(Button)
        onPress={[Function]}
        style={
          Object {
            "backgroundColor": "#ED3E44",
            "flex": 1,
          }
        }
      >
        <Styled(Text)>
          Login
        </Styled(Text)>
      </Styled(Button)>,
      "style": Object {
        "paddingLeft": 15,
        "paddingVertical": 10,
      },
      "virtual": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Styled(Text)>
          Login
        </Styled(Text)>,
        "onPress": [Function],
        "style": Object {
          "backgroundColor": "#ED3E44",
          "flex": 1,
        },
        "virtual": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Login",
          "virtual": undefined,
        },
        "ref": null,
        "rendered": "Login",
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Styled(Button)
          onPress={[Function]}
          style={
            Object {
              "backgroundColor": "#ED3E44",
              "flex": 1,
            }
          }
        >
          <Styled(Text)>
            Login
          </Styled(Text)>
        </Styled(Button)>,
        "style": Object {
          "paddingLeft": 15,
          "paddingVertical": 10,
        },
        "virtual": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Styled(Text)>
            Login
          </Styled(Text)>,
          "onPress": [Function],
          "style": Object {
            "backgroundColor": "#ED3E44",
            "flex": 1,
          },
          "virtual": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Login",
            "virtual": undefined,
          },
          "ref": null,
          "rendered": "Login",
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
